aries_cloudagent/protocols/issue_credential/v2_0/message_types.py

```diff
@@ -43,18 +43,22 @@
     CRED_20_PROPOSAL: {
         V20CredFormat.Format.INDY.api: "hlindy/cred-filter@v2.0",
         V20CredFormat.Format.LD_PROOF.api: "aries/ld-proof-vc-detail@v1.0",
+        V20CredFormat.Format.VC_DI.api: "hlindy/cred-filter@v2.0",
```

Comment:
@tra371 @sarthakvijayvergiya @supersonicwisd1 is this right or should this be didcomm ^

Reply:
This is the issue I mentioned on Discord.  Technically the credential proposal is out of scope, so theres no "didcomm" version of the proposal, however when aca-py issues a credential it creates a "dummy" proposal on-the-fly and attaches it to the credential exchange record.  It then uses this to determine the format of the credential offer, issue, etc.  So even though you specify format "vc_di" the exchange ends up using Indy format.

You can see this if you run the demo:

```
./run_demo faber --wallet-type askar-anoncreds --cred-type vc_di  --events
```

... and ...

```
./run_demo alice --wallet-type askar-anoncreds  --events
```

Then connect the agents and issue a credential, and you can see the format in the resulting credential offer:

```
    "cred_offer": {
...
        "formats": [
            {
                "attach_id": "indy",
                "format": "hlindy/cred-abstract@v2.0"
            }
        ],
...
```

There are two possible fixes for this behaviour:

1. in the V20CredManager.create_offer() method, use the originally requested format, not the format that is attached to the proposal

or

2. Fake out a vc_di format proposal (that looks exactly like the Indy proposal, but with a vc_di format type) so that the create_offer() method picks up the correct type

I think #2 is probably a more elegant solution.  (I know the vc_di proposal was out of scope, but we didn't realize the detail of how it's used in the credential offer flow ...)

Let me know if you have any questions.  I can throw together a quick fix for #2 if you like ...

Reply:
I agree, I'd love for us to do #2, or it doesn't feel like this is "real"

And def would love to see any thoughts/fix you have for it @ianco !

Reply:
I believe the #2 solution is preferred

Reply:
I tried changing the format to:

```
    CRED_20_PROPOSAL: {
        ...,
        V20CredFormat.Format.VC_DI.api: "didcomm/w3c-di-vc-filter@v0.1",
    },
```

(I didn't make any other changes)

... and then I got an error from Alice when receiving the credential:

```
Alice      | 2024-03-08 22:34:36,122 aries_cloudagent.messaging.models.base ERROR V20CredOffer message validation error:
Alice      | Traceback (most recent call last):
Alice      |   File "/home/aries/aries_cloudagent/messaging/models/base.py", line 199, in deserialize
Alice      |     schema.loads(obj) if isinstance(obj, str) else schema.load(obj),
Alice      |   File "/home/aries/.local/lib/python3.9/site-packages/marshmallow/schema.py", line 723, in load
Alice      |     return self._do_load(
Alice      |   File "/home/aries/.local/lib/python3.9/site-packages/marshmallow/schema.py", line 910, in _do_load
Alice      |     raise exc
Alice      | marshmallow.exceptions.ValidationError: {'cred_def_id': ['Unknown field.'], 'nonce': ['Unknown field.'], 'schema_id': ['Unknown field.'], 'key_correctness_proof': ['Unknown field.']}
Alice      | 2024-03-08 22:34:36,123 aries_cloudagent.core.dispatcher ERROR Message parsing failed: Error deserializing message: V20CredOffer schema validation failed, sending problem report
```

With a bit of digging, it looks like the credential offer attachment is wrong, it is being generated as:

```
{
    "schema_id": "DqRJRAnD1es61Xvx4Q99k:2:degree schema:85.58.17",
    "cred_def_id": "DqRJRAnD1es61Xvx4Q99k:3:CL:12:faber.agent.degree_schema",
    "key_correctness_proof":
    {
        ... etc ...
    },
    "nonce": "1032768035276427881713135"
}
```

However the VCDI schema (defined here: https://github.com/hyperledger/aries-cloudagent-python/pull/2830/files#diff-f4aec3b9b074e0ceff1202bfc8f5dd81a387c14d116915fbb178dab705119176R325) doesn't match.

Reply:
It was the format suggest here that was followed 
https://github.com/hyperledger/aries-rfcs/pull/809/files#diff-40b1f86053dd6f0b34250d5be1319d3a0662b96a5a121957fe4dc8cceaa9cbc8R25-R69

Reply:
I think that's the correct format but that's not what the code is producing ...

The schema is defined here, and that's what the received message is getting validated against:  https://github.com/hyperledger/aries-cloudagent-python/pull/2830/files#diff-f4aec3b9b074e0ceff1202bfc8f5dd81a387c14d116915fbb178dab705119176R325

The cred offer is getting created here, and it looks like it's just reusing the Indy offer right now:  https://github.com/hyperledger/aries-cloudagent-python/pull/2830/files#diff-f7d9c8d89f2442a02ac7a5924be4046e195d411791e4cbfd0fd9e7e2ecb987adR211

Reply:
mmm are we sure its getting created here, i can't see where this is called.  maybe its the schema_id if it is called here? is it name magic?

Reply:
ah this was a dupe function i have deleted it